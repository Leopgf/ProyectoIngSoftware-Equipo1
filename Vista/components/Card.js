//IMPORT
import React from 'react';
import { withNavigation } from '@react-navigation/compat';
import PropTypes from 'prop-types';
import { Button as GaButton } from 'galio-framework';
import { StyleSheet, Image, TouchableWithoutFeedback } from 'react-native';
import { Block, Text, theme } from 'galio-framework';
import { nowTheme } from '../constants';

import * as firebase from 'firebase';

//CLASE DE CARD
class Card extends React.Component {
   render() {
    const {
      navigation,
      item,
      horizontal,
      full,
      style,
      ctaColor,
      imageStyle,
      ctaRight,
      titleStyle,
      params,
    } = this.props;

    const imageStyles = [full ? styles.fullImage : styles.horizontalImage, imageStyle];
    const titleStyles = [styles.cardTitle, titleStyle];
    const cardContainer = [styles.card, styles.shadow, style];
    const imgContainer = [
      styles.imageContainer,
      horizontal ? styles.horizontalStyles : styles.verticalStyles,
      styles.shadow,
    ];

    var user = '';

    if(item.userID){
      firebase.firestore().collection('Usuarios').where('usuarioID', '==', item.userID).get().then((usuarios) => {

        usuarios.forEach((usuario) => {
          user = usuario.data().usuario;
        });
      });
    }

    if(item.nombre){
      return (
        <Block row={horizontal} card flex style={cardContainer}>
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
          </TouchableWithoutFeedback>
  
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.nombre && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  )}
                </Block>
                {item.subtitle ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={32}
                      color={nowTheme.COLORS.BLACK}
                    >
                      {item.subtitle}
                    </Text>
                  </Block>
                ) : null}
                {/*DESCRIPCION RECETA*/}
                {item.descripcion && !horizontal ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.descripcion}
                    </Text>
                  </Block>
                ) : null}
                {item.body ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.body}
                    </Text>
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </TouchableWithoutFeedback>
        </Block>
      );
    }else{
        return (
          <Block row={horizontal} card flex style={cardContainer}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
  
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.titulo && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  )}
                </Block>
                {/*DESCRIPCION REVIEW*/}
                {item.mensaje ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.mensaje}
                    </Text>
                  </Block>
                ) : null}
                {item.userID && item.valoracion ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.userID}
                    </Text>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.valoracion}
                    </Text>
                    <GaButton
                          round
                          onlyIcon
                          shadowless
                          icon="star"
                          iconFamily="Font-Awesome"
                          iconColor={'#E63746'}
                          iconSize={nowTheme.SIZES.BASE * 1.4}
                          color={'#FFFFFF'}
                          style={[styles.social, styles.shadow]}
                        />
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </Block>
        );
    }
  }
}

Card.propTypes = {
  item: PropTypes.object,
  horizontal: PropTypes.bool,
  full: PropTypes.bool,
  ctaColor: PropTypes.string,
  imageStyle: PropTypes.any,
  ctaRight: PropTypes.bool,
  titleStyle: PropTypes.any,
  textBodyStyle: PropTypes.any,
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#F2F2F3',
    marginVertical: theme.SIZES.BASE,
    borderWidth: 0,
    minHeight: 114,
    marginBottom: 4,
  },
  cardTitle: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  cardTitle2: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
    color: '#0f1e2e'
  },
  cardDescription: {
    // padding: theme.SIZES.BASE / 2
  },
  imageContainer: {
    borderRadius: 4,
    elevation: 1,
    overflow: 'hidden',
  },
  image: {
    // borderRadius: 3,
  },
  horizontalImage: {
    height: 200,
    width: 'auto',
  },
  horizontalStyles: {
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
  },
  verticalStyles: {
    borderBottomRightRadius: 0,
    borderBottomLeftRadius: 0,
  },
  fullImage: {
    height: 215,
  },
  shadow: {
    shadowColor: '#8898AA',
    shadowOffset: { width: 0, height: 1 },
    shadowRadius: 6,
    shadowOpacity: 0.1,
    elevation: 2,
  },
  articleButton: {
    fontFamily: 'montserrat-bold',
    paddingHorizontal: 9,
    paddingVertical: 7,
  },
});

export default withNavigation(Card);
