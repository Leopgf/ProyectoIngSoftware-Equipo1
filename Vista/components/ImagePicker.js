import React, { useState, useEffect } from 'react';
import { Image, View, Platform, Dimensions } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { StyleSheet } from 'react-native';
import { Button } from 'galio-framework';
import { cloudinary } from 'cloudinary-react';

export default function ImagePickerExample({ onImagePicked, defaultImage }) {
  const { width, height } = Dimensions.get('screen');
  const [image, setImage] = useState(null);

  useEffect(() => {
    (async () => {
      if (Platform.OS !== 'web') {
        const { status } = await ImagePicker.requestCameraRollPermissionsAsync();
        if (status !== 'granted') {
          alert('Sorry, we need camera roll permissions to make this work!');
        }
      }
    })();
  }, []);

  const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    if (!result.cancelled) {
      let file = {
        uri: result.uri,
        type: `test/${result.uri.split('.').pop()}`,
        name: result.uri,
      };

      let apiUrl = 'https://api.cloudinary.com/v1_1/mixosteam/image/upload';

      const formData = new FormData();
      formData.append('file', file);
      formData.append('upload_preset', 'zc9ttas7');
      formData.append('cloud_name', 'mixosteam');

      await fetch(apiUrl, {
        body: formData,
        method: 'POST',
      })
        .then(async (r) => await r.json())
        .then((dato) => {
          console.log('TOY AQUIIII');
          setImage(dato.secure_url);
          onImagePicked({ uri: dato.secure_url });
        })
        .catch((error) => {
          console.log(error);
        });
    }
  };

  const styles = StyleSheet.create({
    smallButton: {
      width: 75,
      height: 28,
    },
    Button: {
      width: width / 2.5,
      height: height / 20,
      shadowOpacity: 0,
    },
  });

  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center',shadowOpacity: 0}}>
      <Button style={styles.Button} onPress={pickImage}>
        Subir imagen
      </Button>
      {image && (
        <Image source={{ uri: image }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
      {defaultImage && !image && (
        <Image source={{ uri: defaultImage }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
    </View>
  );
}
