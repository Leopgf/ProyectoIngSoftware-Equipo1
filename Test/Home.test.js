import React from 'react';
import renderer from 'react-test-renderer';

import { StyleSheet, Dimensions, ScrollView } from "react-native";
import { Block, theme } from "galio-framework";
import { Card } from "../Vista/components";

//CONST
const { width } = Dimensions.get("screen");


describe('<Home />', () => {
  //
    it('renders correctly', () => {
        const tree = renderer.create(
            <ScrollView
            showsVerticalScrollIndicator={false}
            contentContainerStyle={styles.articles}
            >
            <Block flex>
                <Block flex row>
                </Block>
            </Block>
          </ScrollView>
        ).toJSON();
        expect(tree).toMatchSnapshot();
      });
});


//ESTILOS 
const styles = StyleSheet.create({
    home: {
      width: width
    },
    articles: {
      width: width - theme.SIZES.BASE * 2,
      paddingVertical: theme.SIZES.BASE,
      paddingHorizontal: 2,
      fontFamily: 'montserrat-regular'
  
    }
  });