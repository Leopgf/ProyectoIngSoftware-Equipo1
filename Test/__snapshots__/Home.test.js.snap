// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "fontFamily": "montserrat-regular",
      "paddingHorizontal": 2,
      "paddingVertical": 16,
      "width": 718,
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          false,
          Object {
            "flex": 1,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          null,
          false,
          false,
          false,
          null,
          null,
          null,
          undefined,
        ]
      }
      theme={
        Object {
          "COLORS": Object {
            "BLACK": "#000000",
            "BLOCK": "#808080",
            "DRIBBBLE": "#EA4C89",
            "ERROR": "#FE2472",
            "FACEBOOK": "#3B5998",
            "GREY": "#898989",
            "ICON": "#000000",
            "INFO": "#1232FF",
            "INPUT": "#808080",
            "MUTED": "#9FA5AA",
            "NAVBAR": "#F9F9F9",
            "NEUTRAL": "rgba(255,255,255, 0.65)",
            "PLACEHOLDER": "#9FA5AA",
            "PRIMARY": "#B23AFC",
            "SUCCESS": "#45DF31",
            "THEME": "#B23AFC",
            "TRANSPARENT": "transparent",
            "TWITTER": "#5BC0DE",
            "WARNING": "#FF9C09",
            "WHITE": "#FFFFFF",
          },
          "SIZES": Object {
            "ANDROID_ELEVATION": 1,
            "BASE": 16,
            "BLOCK_SHADOW_OPACITY": 0.15,
            "BLOCK_SHADOW_RADIUS": 8,
            "BORDER_RADIUS": 6,
            "BORDER_WIDTH": 0.8,
            "BUTTON_HEIGHT": 44,
            "BUTTON_SHADOW_RADIUS": 10,
            "BUTTON_WIDTH": 144,
            "CARD_AVATAR_HEIGHT": 40,
            "CARD_AVATAR_RADIUS": 20,
            "CARD_AVATAR_WIDTH": 40,
            "CARD_BORDER_RADIUS": 6.4,
            "CARD_BORDER_WIDTH": 0.8,
            "CARD_FOOTER_HORIZONTAL": 12,
            "CARD_FOOTER_VERTICAL": 12,
            "CARD_IMAGE_HEIGHT": 200,
            "CARD_MARGIN_VERTICAL": 14,
            "CARD_ROUND": 3,
            "CARD_ROUNDED": 8,
            "CARD_WIDTH": 718,
            "CHECKBOX_HEIGHT": 20,
            "CHECKBOX_WIDTH": 20,
            "FONT": 16,
            "ICON": 16,
            "INPUT_BORDER_RADIUS": 8,
            "INPUT_BORDER_WIDTH": 0.8,
            "INPUT_HEIGHT": 44,
            "INPUT_HELP_TEXT": 12.8,
            "INPUT_HORIZONTAL": 16,
            "INPUT_LABEL_BOTTOM": 4,
            "INPUT_LABEL_TEXT": 14.4,
            "INPUT_ROUNDED": 27.2,
            "INPUT_TEXT": 14,
            "NAVBAR_HEIGHT": 66,
            "NAVBAR_LEFT_FLEX": 0.5,
            "NAVBAR_LEFT_HEIGHT": 93.38000000000001,
            "NAVBAR_LEFT_MARGIN": 16,
            "NAVBAR_RIGHT_FLEX": 0.5,
            "NAVBAR_RIGHT_HEIGHT": 93.38000000000001,
            "NAVBAR_RIGHT_MARGIN": 16,
            "NAVBAR_TITLE_FLEX": 2,
            "NAVBAR_TITLE_HEIGHT": 93.38000000000001,
            "NAVBAR_TITLE_TEXT": 14,
            "NAVBAR_VERTICAL": 16,
            "OPACITY": 0.8,
            "RADIO_HEIGHT": 24,
            "RADIO_THICKNESS": 2,
            "RADIO_WIDTH": 24,
            "THUMB_SIZE": 25,
            "TRACK_SIZE": 4,
          },
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "flex": 1,
            },
            false,
            false,
            false,
            false,
            false,
            false,
            null,
            false,
            false,
            false,
            null,
            null,
            null,
            undefined,
          ]
        }
        theme={
          Object {
            "COLORS": Object {
              "BLACK": "#000000",
              "BLOCK": "#808080",
              "DRIBBBLE": "#EA4C89",
              "ERROR": "#FE2472",
              "FACEBOOK": "#3B5998",
              "GREY": "#898989",
              "ICON": "#000000",
              "INFO": "#1232FF",
              "INPUT": "#808080",
              "MUTED": "#9FA5AA",
              "NAVBAR": "#F9F9F9",
              "NEUTRAL": "rgba(255,255,255, 0.65)",
              "PLACEHOLDER": "#9FA5AA",
              "PRIMARY": "#B23AFC",
              "SUCCESS": "#45DF31",
              "THEME": "#B23AFC",
              "TRANSPARENT": "transparent",
              "TWITTER": "#5BC0DE",
              "WARNING": "#FF9C09",
              "WHITE": "#FFFFFF",
            },
            "SIZES": Object {
              "ANDROID_ELEVATION": 1,
              "BASE": 16,
              "BLOCK_SHADOW_OPACITY": 0.15,
              "BLOCK_SHADOW_RADIUS": 8,
              "BORDER_RADIUS": 6,
              "BORDER_WIDTH": 0.8,
              "BUTTON_HEIGHT": 44,
              "BUTTON_SHADOW_RADIUS": 10,
              "BUTTON_WIDTH": 144,
              "CARD_AVATAR_HEIGHT": 40,
              "CARD_AVATAR_RADIUS": 20,
              "CARD_AVATAR_WIDTH": 40,
              "CARD_BORDER_RADIUS": 6.4,
              "CARD_BORDER_WIDTH": 0.8,
              "CARD_FOOTER_HORIZONTAL": 12,
              "CARD_FOOTER_VERTICAL": 12,
              "CARD_IMAGE_HEIGHT": 200,
              "CARD_MARGIN_VERTICAL": 14,
              "CARD_ROUND": 3,
              "CARD_ROUNDED": 8,
              "CARD_WIDTH": 718,
              "CHECKBOX_HEIGHT": 20,
              "CHECKBOX_WIDTH": 20,
              "FONT": 16,
              "ICON": 16,
              "INPUT_BORDER_RADIUS": 8,
              "INPUT_BORDER_WIDTH": 0.8,
              "INPUT_HEIGHT": 44,
              "INPUT_HELP_TEXT": 12.8,
              "INPUT_HORIZONTAL": 16,
              "INPUT_LABEL_BOTTOM": 4,
              "INPUT_LABEL_TEXT": 14.4,
              "INPUT_ROUNDED": 27.2,
              "INPUT_TEXT": 14,
              "NAVBAR_HEIGHT": 66,
              "NAVBAR_LEFT_FLEX": 0.5,
              "NAVBAR_LEFT_HEIGHT": 93.38000000000001,
              "NAVBAR_LEFT_MARGIN": 16,
              "NAVBAR_RIGHT_FLEX": 0.5,
              "NAVBAR_RIGHT_HEIGHT": 93.38000000000001,
              "NAVBAR_RIGHT_MARGIN": 16,
              "NAVBAR_TITLE_FLEX": 2,
              "NAVBAR_TITLE_HEIGHT": 93.38000000000001,
              "NAVBAR_TITLE_TEXT": 14,
              "NAVBAR_VERTICAL": 16,
              "OPACITY": 0.8,
              "RADIO_HEIGHT": 24,
              "RADIO_THICKNESS": 2,
              "RADIO_WIDTH": 24,
              "THUMB_SIZE": 25,
              "TRACK_SIZE": 4,
            },
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
